Core/include/Acts/Propagator/AtlasStepper.hpp

```diff
@@ -558,12 +558,16 @@ class AtlasStepper {
   ///
   /// @param [in] state State that will be presented as @c BoundState
   /// @param [in] surface The surface to which we bind the state
+  /// @param [in] transportCov Flag steering covariance transport
+  /// @param [in[ attachCov Decision whether the covariance should be attached
```

Comment:
Is there a case that we don't want to attach the covariance to the track parameters?

Reply:
In case we don't have a covariance matrix then it should not be attached, I guess. I mean you can attach it but it would be a zero matrix and it becomes necessary to test for that case. But in general, there is no strong argument, I just wanted to keep the convention as it is.

Reply:
I see. Another option might be that, we initialize the `cov` with the `covarianceMatrix` (instead of `std::nullopt`) and only update it if the `covTransport` is true?

Reply:
I did this in an earlier commit. It is possible but would become quite ugly since you have to set this flag twice in the (C)KF. Once before the state call and once afterwards. If I use the flag of the state as "transport in general" and the one in the method as "transport in this particular case" then it would have a meaning behind it.

Reply:
flag is removed